src/components/search/SearchService.js

```diff
@@ -100,31 +110,21 @@ goog.provide('ga_search_service');
           position =
             [left > right ? left : right,
               right < left ? right : left];
-          if (ol.extent.containsCoordinate(
-              extent, position)) {
-            valid = true;
+          if (ol.extent.containsCoordinate(extent, position)) {
+            defer.resolve(roundCoordinates(position));
           } else {
-            position = ol.proj.transform(position,
-              'EPSG:2056', 'EPSG:21781');
-            if (ol.extent.containsCoordinate(
-                extent, position)) {
-              valid = true;
-            } else {
-              position =
-                [left < right ? left : right,
-                  right > left ? right : left];
-              position = ol.proj.transform(position,
-                'EPSG:4326', 'EPSG:21781');
-              if (ol.extent.containsCoordinate(
-                extent, position)) {
-                valid = true;
+            return gaReframe.get95To03(position).then(function(position) {
+              if (ol.extent.containsCoordinate(extent, position)) {
+                return roundCoordinates(position);
               }
-            }
+            });
           }
+        } else if (valid) {
+          defer.resolve(roundCoordinates(position));
+        } else {
+          defer.resolve(undefined);
```

Comment:
Just thinking that  for the sake of clarity we coulduse  `$q.when(the value)` every where instead of the `defer.resolve` variable.